src/ameedf/jb8223/assignments/hw02/Clock.java

```diff
@@ -34,4 +34,79 @@
  * BONUS 2: the same functionality as in "BONUS 1" while using as few member variables / memory as possible
  */
 public class Clock {
+	private byte hours;
+	private byte minutes;
+	private byte seconds;
+	private short milliseconds;
+	
+	//C-tor initialize time to 14:07:53.980
+	public Clock() {
+        this.hours = 14;
+        this.minutes = 7;
+        this.seconds = 53;
+        this.milliseconds = 980;
+    }
+	
+	//C-tor initialize time by user input
+	public Clock(int hours, int minutes, int seconds, int milliseconds) {
+        this.hours = (byte) hours;
+        this.minutes = (byte) minutes;
+        this.seconds = (byte) seconds;
+        this.milliseconds = (byte) milliseconds;
+    }
+
+	public void addHours(int hours) {
+		if(hours > 0 && hours < 1000000000 ) {
+			this.hours = (byte) ((this.hours + hours) % 24);
+		}
+	}
+	
+	public void addMinutes(int minutes) {
```

Comment:
bug found: clock = 16:49:05.065 --> adding 30 minutes: 16:19:05.065 (should've been 17:19:05.065)

Reply:
Yes, sorry, fixed.

Don't commit to production after midnight.. ))

Reply:
:)